src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -89,10 +89,11 @@ public function setValue(&$objectOrArray, $propertyPath, $value)
                 } else {
                     $this->writeProperty($objectOrArray, $property, $value);
                 }
+
+                $overwrite = !$propertyValues[$i][self::IS_REF];
```

Comment:
it should break the loop when not needed anymore. currently it will loop without doing anything.